paddle/framework/tensor.h

```diff
@@ -87,26 +87,28 @@ class Tensor {
   /**
    * @brief   Copy the content of external tensor to a new place.
    *
-   * @param[in] src   The external tensor.
-   * @param[in] ctx   The device context contains place where to store.
+   * @param[in] src        The external tensor.
+   * @param[in] dst_place  The dst place.
+   * @param[in] ctx        The device context contains CUDA stream.
    *
    * @note    CopyFrom supports CPU <-> GPU, GPU <-> GPU.
    */
   template <typename T>
-  inline void CopyFrom(const Tensor& src, const platform::Place& dst_place);
+  inline void CopyFrom(const Tensor& src, const platform::Place& dst_place,
+                       const platform::DeviceContext& ctx);
 
   /**
    * @brief   Copy the content of an external vector to a tensor.
    *
-   * @param[in] src   The external vector.
-   * @param[in] ctx   The device context contains place where to store.
+   * @param[in] src        The external tensor.
+   * @param[in] ctx        The device context contains CUDA stream.
```

Comment:
Suppose that we need to copy from GPU to CPU, what kind of DeviceContext here do we need -- CPUContext or CUDAContext? What if the reverse order -- from GPU to CPU?  What if between CPU and FPGA?

Reply:
When we make Copy between devices, we have to pass a DeviceContext which contains device resources. In GPU, the device resource is CUDA stream. Copying from GPU to CPU, CPU to GPU and GPU to GPU all need a CUDA stream.